### PR TITLE
Rework types documentation

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/expr/types.py [hunk 1]
```diff
@@ -278,14 +263,9 @@ def _repr(self):
...
-    """
-    Hail type corresponding to str.
-
-    .. include:: ../_templates/hailType.rst
-
-    - `expression language documentation <types.html#string>`__
-    - in Python, these are represented natively as Python unicode strings
+    """Hail type for text strings.
 
+    In Python, these are represented as :obj:`unicode` strings.
```
Comment (danking) [inline python/hail/expr/types.py]: In Python 3 (which, AFAIK, is the target python for the release of 0.2), this is ``` :obj:`str` ``` right?

File: python/hail/expr/types.py [hunk 2]
```diff
@@ -310,14 +290,9 @@ def _repr(self):
...
 
 class TBoolean(Type):
-    """
-    Hail type corresponding to bool.
-
-    .. include:: ../_templates/hailType.rst
-
-    - `expression language documentation <types.html#boolean>`__
-    - in Python, these are represented natively as Python booleans (i.e. ``True`` and ``False``)
+    """Hail type for boolean (``True`` or ``False``) values.
```
Comment (danking) [inline python/hail/expr/types.py]: pedantic, but boolean is named for a person, so I kind of feel it should be capitalized in English text.

Comment (tpoterba) [reply python/hail/expr/types.py]: Really? Happy to change but scientific units like newton and joule aren't usually capitalized

Comment (danking) [reply python/hail/expr/types.py]: Hmm. english.se [seems to prefer capitalization of Boolean](https://english.stackexchange.com/questions/4481/should-the-word-boolean-be-capitalized) in particular, though there's a telling comment about the decapitalization of words over time re: Abelian.

(Aside: I had no idea that newton was meant to be lower-case, I've always capitalized it in Physics class following the same logic as Boolean!)

I don't have a strong feeling on Boolean either way :)

**Pull request merged**